flink-connector-pulsar/src/test/java/org/apache/flink/connector/pulsar/common/handler/PulsarAdminInvocationHandlerTest.java

```diff
@@ -0,0 +1,309 @@
+import org.apache.pulsar.client.admin.ProxyStats;
+import org.apache.pulsar.client.admin.PulsarAdmin;
+import org.apache.pulsar.client.admin.PulsarAdminException;
+import org.apache.pulsar.client.admin.PulsarAdminException.NotFoundException;
+import org.apache.pulsar.client.admin.ResourceGroups;
+import org.apache.pulsar.client.admin.ResourceQuotas;
+import org.apache.pulsar.client.admin.Schemas;
+import org.apache.pulsar.client.admin.Sink;
+import org.apache.pulsar.client.admin.Sinks;
+import org.apache.pulsar.client.admin.Source;
+import org.apache.pulsar.client.admin.Sources;
+import org.apache.pulsar.client.admin.Tenants;
+import org.apache.pulsar.client.admin.TopicPolicies;
+import org.apache.pulsar.client.admin.Topics;
+import org.apache.pulsar.client.admin.Transactions;
+import org.apache.pulsar.client.admin.Worker;
+import org.junit.jupiter.api.Test;
+
+import java.lang.reflect.Proxy;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ThreadLocalRandom;
+
+import static org.apache.commons.lang3.RandomStringUtils.randomAlphanumeric;
+import static org.apache.flink.connector.pulsar.common.config.PulsarOptions.PULSAR_ADMIN_REQUEST_RATES;
+import static org.apache.flink.connector.pulsar.common.config.PulsarOptions.PULSAR_ADMIN_REQUEST_RETRIES;
+import static org.apache.flink.connector.pulsar.common.config.PulsarOptions.PULSAR_ADMIN_REQUEST_WAIT_MILLIS;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+/** Unit test of {@link PulsarAdminInvocationHandler}. */
+class PulsarAdminInvocationHandlerTest {
+
+    @Test
+    void retryWithFixedTimesOnRequestFailure() {
+        int retryTimes = 5 + ThreadLocalRandom.current().nextInt(5);
+        String errorMsg = "Always failed with reason: " + randomAlphanumeric(10);
+
+        PulsarAdminTestImpl admin = new PulsarAdminTestImpl(errorMsg);
+
+        Configuration configuration = new Configuration();
+        configuration.set(PULSAR_ADMIN_REQUEST_RETRIES, retryTimes);
+        configuration.set(PULSAR_ADMIN_REQUEST_WAIT_MILLIS, 50L);
+        configuration.set(PULSAR_ADMIN_REQUEST_RATES, 1000);
+        SinkConfiguration sinkConfiguration = new SinkConfiguration(configuration);
+
+        PulsarAdminInvocationHandler handler =
+                new PulsarAdminInvocationHandler(admin, sinkConfiguration);
+
+        PulsarAdmin proxyAdmin =
+                (PulsarAdmin)
+                        Proxy.newProxyInstance(
+                                PulsarAdmin.class.getClassLoader(),
+                                new Class[] {PulsarAdmin.class},
+                                handler);
+
+        assertThatThrownBy(() -> proxyAdmin.lookups().lookupPartitionedTopic("aa"))
+                .isInstanceOf(PulsarAdminException.class)
+                .hasMessage(errorMsg);
+        assertEquals(retryTimes, admin.lookup.callingTimes());
```

Comment:
```suggestion
        assertThat(admin.lookup.callingTimes()).isEqualTo(retryTimes);
```

Reply:
Since there are a lot of JUnit 5 assertions, can we start a new PR which replaces all the assertions to AssertJ and keep this as it is?

<img width="614" alt="image" src="https://user-images.githubusercontent.com/1761698/219316403-4ecdb1a9-9839-4a2d-b4c8-5363ffc4a97d.png">

Reply:
I think this may be a tech debt like the previous PR. https://github.com/apache/flink-connector-pulsar/pull/24

Reply:
Make sense to me.